### ISSUE TITLE
study blog

### ISSUE THREAD

**Issue by SailerNote** [SailerNote/To-do-list#7]
帮助我解决问题的博客，感觉各位的分享与奉献。

**Comment by SailerNote** [comment]
[Django: 使用 Q 对象构建复杂的查询语句](https://mozillazg.github.io/2015/11/django-the-power-of-q-objects-and-how-to-use-q-object.html)

**Comment by SailerNote** [comment]
[Django sql in查询](http://www.360doc.com/content/14/0424/20/16044571_371812557.shtml)
django filter的查询字段中并没有__notin这种形式，如果想讲查询的排除掉则使用exclude替换filter
目标
`ModelName.objects.filter(attribute__notin=attrobute)` 
实现
`ModelName.objects.exclude(attribute__in=attribute)`

**Comment by SailerNote** [comment]
[从数据库反向生成django的models](http://www.yihaomen.com/article/python/244.htm)
在接手一些二次开发的项目，需要建立在原有数据库结构的基础上。用反向生成可以*快速，准确*的完成目的。

**Comment by SailerNote** [comment]
[python super() 漩涡鸣人的博客](http://www.cnblogs.com/lovemo1314/archive/2011/05/03/2035005.html)
python的继承遵循ORM原则，印象里还看过一篇更精细的文章，从图论和继承的弊病又来解决了这个问题，不断的0入1出寻找节点。不过这篇是我的启蒙文章，找到以后再加进来另一篇。

**Comment by SailerNote** [comment]
[微信公众号开发只模板消息](http://www.jianshu.com/p/eb0e9c4dcdfe)
在官方文档不完善的情况下，自己尝试是一件很浪费时间的事情。能有人把坑趟完感激的很。顺便吐槽一些软件，程序更新后文档和开发模式都会变更，那些很细心的博客也失去效果。

**Comment by SailerNote** [comment]
[python 时间 日期 时间戳处理](http://blog.csdn.net/xiaobing_blog/article/details/12591917)

**Comment by SailerNote** [comment]
[django-crontab 实现django定时运行函数](https://www.zhidaow.com/post/django-crontab)

**Comment by SailerNote** [comment]
python中根据字典的键或者值对字典排序

[How to sort a Python dict (dictionary) by keys or values](https://www.saltycrane.com/blog/2007/09/how-to-sort-python-dictionary-by-keys/)

**Comment by SailerNote** [comment]
django imagefield 上传图片问题

[关于Django中访问通过ImageField上传的图片的问题](https://segmentfault.com/q/1010000004122558)

这里有个小坑就是，上传的文件是要放在MEDIA_URL，而不是STATIC_URL中的。配置的时候需要注意。

**Comment by SailerNote** [comment]
django 定时任务

[张亚楠博客 DJANGO-CRONTAB实现DJANGO定时任务](https://www.zhidaow.com/post/django-crontab)

小型的业务可以使用这个定时执行任务的类库，很方便，缺点是不支持windows。不过还是建议学习一下celery，定制的效果和扩展要优秀的多。

**Comment by SailerNote** [comment]
如何让IIS7 显示详细错误信息

[让IIS 7显示ASP的详细错误信息](http://zhangqian.me/iis-7-error/)

感觉windows上部署服务器还是很复杂，遇到问题解决方案也不多。更多是图文的攻略，而不是脚本命令。

**Comment by SailerNote** [comment]
正则表达式 贪婪模式和非贪婪模式

[正则基础之——贪婪与非贪婪模式](http://blog.csdn.net/lxcnn/article/details/4756030)

简单讲，就是在一个字符串中起点一致的长度分别为3和5的两个子串都符合正则，非贪婪模式匹配到3就停止了。而贪婪模式会继续向后尝试。找到5为止。文中的例子更形象，一看就明白了。

**Comment by SailerNote** [comment]
[python 代码自动生成的方法 （代码生成器）](http://ju.outofmemory.cn/entry/85205)

django 在rest framework 框架下mvc和restful风格都实现的不错，很多基础业务代码很相似。
仍然不知道大佬是怎么实现快速生成的，按理说使用类库更好。这篇文章启发了我最后使用了jinja2这种前端模板用来生成python代码，效果还可以。

**Comment by SailerNote** [comment]
[Python 优雅的操作字典](https://www.linuxzen.com/python-you-ya-de-cao-zuo-zi-dian.html)

对python 字典操作的总结，虽然说这种东西还是看官方文档最好。我主要喜欢博主的排版。

**Comment by SailerNote** [comment]
[Python 3 Module of the Week](https://pymotw.com/3/)

python3中各种组件的使用，非常详细而已例子丰富完整。跟着做下来收益颇丰。很适合python初学者。

**Comment by SailerNote** [comment]
[python-docx](https://python-docx.readthedocs.io/en/latest/index.html#)

创建和更新 Microsoft Word，没有实践使用过。这个需求后来被取消变成上传下载文件了。

**Comment by SailerNote** [comment]
[Python命名空间和作用域窥探](http://blog.cipherc.com/2015/04/25/python_namespace_and_scope/)

很详细推荐一下

**Comment by SailerNote** [comment]
[柳婼 の blog](https://www.liuchuo.net/)

最开始积累算法知识时候经常逛的博客，会有题目和答案，以及详解。支持网站还捐了杯咖啡。

**Comment by SailerNote** [comment]
Boom filter算法
[大规模数据查重的多种方法，及BLOOM FILTER的应用](http://www.blogjava.net/paulwong/archive/2013/02/06/394980.html)
如果学习爬虫的话，必修。

**Comment by SailerNote** [comment]
[django annotate and count: how to filter the ones to include in count](https://stackoverflow.com/questions/29440374/django-annotate-and-count-how-to-filter-the-ones-to-include-in-count)

应该是在一个数据库设计有问题的情况下强行聚合产生的问题。已经不记得具体的情况了，留下来仅作参考。

**Comment by SailerNote** [comment]
[Returning nearby locations in Django](https://stackoverflow.com/questions/25623829/returning-nearby-locations-in-django)

计算附近的人使用的数据库查询语句。在django中可以使用。

**Comment by SailerNote** [comment]
[八大排序算法的Python实现](http://www.imooc.com/article/3927)

**Comment by SailerNote** [comment]
[Overleaf](https://www.overleaf.com/)

如果你是一个Latex的使用者，这个网站可以提供非常丰富的模板。

**Comment by SailerNote** [comment]
[【不周山之读薄 CSAPP】](http://wdxtub.com/2016/04/16/thin-csapp-0/)

CSAPP导读文章，看书的时候可以一起看。一些不理解的地方可能会得到灵感。

**Comment by SailerNote** [comment]
[Transactions in REST?](https://stackoverflow.com/questions/147207/transactions-in-rest)

关于restful和事务处理的讨论，

另外据说google的bigtable数据库是不使用事务的。

**Comment by SailerNote** [comment]
[Django REST Framework: Using the request object](http://masnun.rocks/2017/03/27/django-rest-framework-using-request-object/)

这篇的干货：request中数据可以传到serializer中使用。

**Comment by SailerNote** [comment]
[TAOCP](http://www-cs-faculty.stanford.edu/~knuth/taocp.html)


The Art of Computer Programming (TAOCP) 又是一本没听过的神书。
--

**Comment by SailerNote** [comment]
[多范式编程语言](http://blog.csdn.net/hnyzyty/article/details/49887259)

科普文章

**Comment by SailerNote** [comment]
[C++中Overload、Overwrite及Override的区别](http://blog.csdn.net/penzo/article/details/6001193)

**Comment by SailerNote** [comment]
[排序应该在数据库还是在应用程序中进行？](http://blog.csdn.net/truong/article/details/10499699)

就我而言，上量的数据，我自认比不上优化的数据库。所以老老实实用ORM

**Comment by SailerNote** [comment]
[miki725/django-rest-framework-bulk](https://github.com/miki725/django-rest-framework-bulk/)

django 批量处理的类库，本质是判断参数类型如果是list的话循环object的viewset，并且添加参数mant=True。主要是开阔思路。感受一些大佬一劳解决问的方式，源码很少，对比一些DRF的viewset可以学到不少。

**Comment by SailerNote** [comment]
源于代码应该放在docker外还是docker内部的问题

[请教下代码放在Docker里面还是外面呢](http://dockone.io/question/24)

这里给出的解决建议是 代码不放入镜像 但挂载在容器中。

**Comment by SailerNote** [comment]
DBeaver

[ubuntu 16.04 LTS安装DBeaver](http://www.linuxdiyf.com/linux/28943.html)

linux下可视数据库管理工具。挺好用的。

**Comment by SailerNote** [comment]
django 时区问题

[Django时区问题](http://www.jianshu.com/p/f0d49fac155d)

！ 划重点。这个问题困扰了我很久。为什么时间总是UTC的问题。

**Comment by SailerNote** [comment]
Lisp 的本质

[Lisp的本质(The Nature of Lisp)](https://www.cnblogs.com/Leap-abead/articles/762180.html)

科普文章但是太干了。不过看着就写的很好的样子，希望有朝一日能看懂。

**Comment by SailerNote** [comment]
用于django 的MS SQL server 数据库引擎

[michiya/django-pyodbc-azure](https://github.com/michiya/django-pyodbc-azure)

**Comment by SailerNote** [comment]
python 处理word

[用python处理MS Word](http://blog.csdn.net/qianchenglenger/article/details/51582005)

**Comment by SailerNote** [comment]
stack overflow 2016 年架构分析
[Stack Overflow: The Architecture - 2016 Edition](https://nickcraver.com/blog/2016/02/17/stack-overflow-the-architecture-2016-edition/)

又是干货

**Comment by SailerNote** [comment]
准确率 召回率 F值
[推荐系统评测指标—准确率(Precision)、召回率(Recall)、F值(F-Measure) ](http://bookshadow.com/weblog/2014/06/10/precision-recall-f-measure/)

**Comment by SailerNote** [comment]
训练集 验证集 测试集
[[综] 训练集(train set) 验证集(validation set) 测试集(test set)](http://www.cnblogs.com/xfzhang/archive/2013/05/24/3096412.html)

**Comment by SailerNote** [comment]
stack overflow上关于python问题的整理

[stackoverflow-about-python](https://taizilongxu.gitbooks.io/stackoverflow-about-python/content/index.html)

问题和解答都很有价值，但是解答本身有很多不是很容易理解。建议当作索引来看。

**Comment by SailerNote** [comment]
[同步，异步，阻塞，非阻塞等关系轻松理解 ](https://github.com/calidion/calidion.github.io/issues/40)

**Comment by SailerNote** [comment]
初学git可以看看

[git - 简明指南](http://rogerdudler.github.io/git-guide/index.zh.html)

画风颜色比较好看，学起来不会无聊

**Comment by SailerNote** [comment]
机器学习初步学习笔记
[机器学习初步学习笔记](http://www.cnblogs.com/NaughtyBaby/p/5625815.html)

**Comment by SailerNote** [comment]
处理django rest framework  permission_classes中的复杂权限。

[caxap/rest_condition](https://github.com/caxap/rest_condition)

**Comment by SailerNote** [comment]
为什么python中没有switch case?
[Why isn’t there a switch or case statement in Python?](https://docs.python.org/2/faq/design.html#why-isn-t-there-a-switch-or-case-statement-in-python)

**Comment by SailerNote** [comment]
[django celery 定时任务](https://zhuanlan.zhihu.com/p/30742764)

**Comment by SailerNote** [comment]
git 修改远程仓库地址。
[git 修改远程仓库地址](https://ddnode.com/2015/04/14/git-modify-remote-responsity-url.html)

当项目切换地址或者更换Group的时候url会变更。远程和本地都需要更新
